lib/rucio/core/rule.py

```diff
@@ -890,12 +1027,20 @@ def list_associated_rules_for_file(scope, name, *, session: "Session"):
     """
 
     rucio.core.did.get_did(scope=scope, name=name, session=session)  # Check if the did acually exists
-    query = session.query(models.ReplicationRule).\
-        with_hint(models.ReplicaLock, "INDEX(LOCKS LOCKS_PK)", 'oracle').\
-        join(models.ReplicaLock, models.ReplicationRule.id == models.ReplicaLock.rule_id).\
-        filter(models.ReplicaLock.scope == scope, models.ReplicaLock.name == name).distinct()
+    stmt = select(
+        models.ReplicationRule
+    ).with_hint(
+        models.ReplicationRule, "INDEX(RULES RULES_PK)", 'oracle'
+    ).join(
+        models.ReplicaLock,
+        models.ReplicationRule.id == models.ReplicaLock.rule_id
+    ).where(
+        and_(models.ReplicaLock.scope == scope,
+             models.ReplicaLock.name == name)
+    ).distinct()
+
     try:
-        for rule in query.yield_per(5):
+        for rule in session.execute(stmt).yield_per(5).scalars():
```

Comment:
I wonder if .scalars() should be moved to the next line. Thus if the generator works like this correctly

Reply:
I am uncertain about how this is handled internally, will write a test script and figure it out. However, scalars are apparently not needed here, as the result is unpacked regardless. Removed .scalars() and returning

`rule.to_dict()`